core/sql/executor/ExSequence.cpp

```diff
@@ -1487,7 +1487,35 @@ void ExSequenceTcb::updateDiagsArea(  ExeErrorCode rc_)
     }
     if (!da->contains((Lng32) -rc_))
     {
-      *da << DgSqlCode(-rc_);
+      char msg[512];
```

Comment:
It would be nice if you can encapsulate these changes in cluster_->flush(). The caller doesn't need to know  about the internals of cluster_